src/test/java/org/dmfs/oauth2/client/tokens/JsonAccessTokenTest.java

```diff
@@ -51,4 +52,26 @@ public void testScope() throws Exception
         assertThat(new JsonAccessToken(jsonObject, dummyScope).scope(), Matchers.<OAuth2Scope>is(new StringScope("scope1 scope2")));
     }
 
+    @Test
+    public void testCustomPayload() throws Exception
+    {
+        OAuth2Scope dummyScope = dummy(OAuth2Scope.class);
+        JSONObject jsonObject = new JSONObject("{\"idToken\":\"id_token_value\"}");
+
+        Optional<CharSequence> parameter = new JsonAccessToken(jsonObject, dummyScope).extraParameter("idToken");
+        assertThat(parameter.isPresent(), Matchers.is(true));
+        assertThat(parameter.value().toString(), Matchers.is("id_token_value"));
```

Comment:
using our `present` Matcher you can do the entire test in a single statement:

```java
assertThat(
    new JsonAccessToken(new JSONObject("{\"idToken\":\"id_token_value\"}"), dummy(OAuth2Scope.class))
        .extraParameter("idToken"),
    is(present("id_token_value")));
```

I'm a big fan of this style because it's declarative and has less noise.

For the absent test below you can use the `absent` Matcher.

Reply:
To clarify before I commit my change, you would prefer declaring the `JSONObject` and `OAuth2Scope` inline with the declaration of `JsonAccessToken`, right?

Reply:
Yes, absolutely.